genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/JobExecutionStateMachineImpl.java

```diff
@@ -78,4 +78,13 @@ public States waitForStop() throws InterruptedException {
             throw e;
         }
     }
+
+    @Override
+    public void stop() {
+        log.info("Stopping state machine (in state: {})", stateMachine.getState().getId());
+        // This event is processed iff the state machine has not reached the state where the job is launched.
```

Comment:
`iff`

Reply:
That was intentional (https://en.wikipedia.org/wiki/If_and_only_if)